internal/db/nosql/cassandra/option.go

```diff
@@ -125,23 +139,32 @@ func WithConnectTimeout(dur string) Option {
 	}
 }
 
+// WithPort returns the option to set the port number
 func WithPort(port int) Option {
 	return func(c *client) error {
-		c.port = port
+		if port > 0 {
+			c.port = port
+		}
 		return nil
 	}
 }
 
+// WithKeyspace returns the option to set the keyspace
 func WithKeyspace(keyspace string) Option {
 	return func(c *client) error {
-		c.keyspace = keyspace
+		if len(keyspace) != 0 {
+			c.keyspace = keyspace
+		}
 		return nil
 	}
 }
 
+// WithNumConns returns the option to set the number of connection per host
 func WithNumConns(numConns int) Option {
 	return func(c *client) error {
-		c.numConns = numConns
+		if numConns > 0 {
+			c.numConns = numConns
+		}
```

Comment:
need to authors' review

Reply:
see
https://github.com/gocql/gocql/blob/master/connectionpool.go#L278-L341
https://github.com/gocql/gocql/blob/master/cluster.go#L53

numConns always should be >= 0 but gocql recommends 2 as default value, so we have to validate numConns >= 0 and we should set defaultOption & charts values.yaml

Reply:
it has already been set on the defaultOption and helm chart here
https://github.com/vdaas/vald/blob/2348bba721f6a77de50e8aada31fe9af1681dd36/internal/db/nosql/cassandra/option.go#L50
https://github.com/vdaas/vald/blob/6ef767593061959e6045ff92f4f836cef7c9db27/charts/vald/values.yaml#L2313